Core.ApplicationServices/Organizations/OrganizationRightsService.cs

```diff
@@ -75,6 +87,122 @@ public class OrganizationRightsService : IOrganizationRightsService
             return RemoveRight(right);
         }
 
+        public Maybe<OperationError> RemoveUnitRightsByIds(Guid organizationUuid, Guid unitUuid, IEnumerable<int> rightIds)
+        {
+            using var transaction = _transactionManager.Begin();
+
+            var unitResult = GetOrganizationUnitByUuid(organizationUuid, unitUuid);
+            if (unitResult.Failed)
+            {
+                return unitResult.Error;
+            }
+            
+            var rightsToDelete = new List<OrganizationUnitRight>();
+            foreach (var rightId in rightIds)
+            {
+                var unitRightResult = unitResult.Value.GetRight(rightId);
+                if (unitRightResult.Failed)
+                {
+                    return unitRightResult.Error;
+                }
+                
+                rightsToDelete.Add(unitRightResult.Value);
+            }
+            var userIds = rightsToDelete.Select(x => x.UserId).ToList();
+
+            _unitRightRepository.RemoveRange(rightsToDelete);
+            foreach (var userId in userIds.Distinct())
+            {
+                _domainEvents.Raise(new AdministrativeAccessRightsChanged(userId));
+            }
+            _unitRightRepository.Save();
+            transaction.Commit();
+
+            return Maybe<OperationError>.None;
+        }
+
+        public Maybe<OperationError> TransferUnitRightsByIds(Guid organizationUuid, Guid unitUuid, Guid targetUnitUuid, IEnumerable<int> rightIds)
+        {
+            using var transaction = _transactionManager.Begin();
+
+            var organizationResult = GetOrganizationAndAuthorizeModification(organizationUuid);
+            if (organizationResult.Failed)
+            {
+                return organizationResult.Error;
+            }
+
+            var unitResult = GetOrganizationUnit(organizationResult.Value, unitUuid);
```

Comment:
you should "unbox" the "Value" instead of ".Value"... all the time either using a "Bind" on the GetOrganizationModification or by unboxing it to a var